src/params.rs

```diff
@@ -61,6 +62,26 @@ pub struct RunParameters {
                             // HOME: /
 }
 
+impl RunParameters {
+    /// Examines the local network interfaces, and tries to find our assigned IP within the data network.
+    /// If running in a sidecar-less environment, the loopback address is returned.
+    pub fn data_network_ip(&self) -> std::io::Result<Option<IpAddr>> {
+        if !self.test_sidecar {
+            // this must be a local:exec runner and we currently don't support traffic shaping on it for now, just return the loopback address.
+            return Ok(Some(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1))));
+        }
+
+        for interface in if_addrs::get_if_addrs()? {
+            let ip = interface.addr.ip();
+            if self.test_subnet.contains(ip) {
+                return Ok(Some(ip));
+            }
+        }
```

Comment:
```suggestion
    Ok(if_addrs::get_if_addrs()?.into_iter().map(|i| i.addr.ip()).find(|ip| self.test_subnet.contains(ip)))
```

Would using `Iterator` work here as well?

Reply:
Cool, thank you!